tools/prepare-release.js

```diff
@@ -0,0 +1,305 @@
+ */
+
+const { Octokit } = require("@octokit/rest");
+const fs = require("fs");
+const path = require("path");
+const os = require("os");
+const { execSync } = require("child_process");
+const rimraf = require("rimraf");
+
+// Repository to process and PR reviewers
+const owner = "w3c";
+const repo = "webref";
+const reviewers = ["dontcallmedom", "foolip", "tidoust"];
+
+// Current package version in npm
+let latestReleasedVersion = "";
+
+
+function btoa(str) {
+  return Buffer.from(str).toString("base64");
+}
+
+
+/**
+ * Compute diff between the released npm package and the contents of the repo
+ *
+ * @function
+ * @param {String} type Package name. One of "css" or "idl"
+ * @return {String} The results of running the diff. An empty string if contents
+ *   match.
+ */
+function computeDiff(type) {
+  // Install @webref package in tmp folder
+  const tmpFolder = fs.mkdtempSync(path.join(os.tmpdir(), "webref-"));
+  execSync(`npm install @webref/${type}`, {
+    cwd: tmpFolder
+  });
+
+  // Extract released version (will be used in the body of the pre-release PR)
+  latestReleasedVersion = require(path.join(tmpFolder, "node_modules", "@webref", type, "package.json")).version;
+
+  // Diff does not take the package.json file into account because "npm install"
+  // adds properties that start with "_" to that file which do not exist in the
+  // repo version of the file.
+  // The "diff" command exits with a non zero status code when there is a diff,
+  // which would throw an exception. Final "echo" command turns that status
+  // code to 0 to avoid the exception.
+  const installedIdl = path.join(tmpFolder, "node_modules", "@webref", type);
+  let diff = execSync(
+    `diff ${installedIdl} packages/${type} --ignore-trailing-space --exclude=package.json --unified=3 || echo -n`,
+    { encoding: "utf8" });
+
+  // Clean up tmp folder
+  try {
+    rimraf.sync(tmpFolder);
+  }
+  catch {
+  }
+
+  // On Windows, the "echo -n" command actually outputs "-n"
+  // Also replace tmp folder name in diff to make it possible to compare diff
+  // with diff that may already exist in a pending pre-release PR.
+  // Note the diff command doubles backslashes, which also need to be doubled
+  // in the string passed to RegExp for them not to be interpreted (In practice,
+  // these backslashes also only appear on Windows)
+  diff = diff
+    .replace(/\-n\s*$/, "")
+    .replace(new RegExp(tmpFolder.replace(/\\/g, "\\\\\\\\"), "g"), "webref");
+
+  return diff;
+}
+
+
+/**
+ * Create or update pre-release pull request
+ *
+ * @function
+ * @param {String} type Package name. One of "css" or "idl"
+ */
+async function prepareRelease(type) {
+  // Compute a reasonably unique ID
+  const uid = (new Date()).toISOString().replace(/[\-T:\.Z]/g, "");
+
+  console.log("Look for a pending pre-release PR");
+  const searchResponse = await octokit.search.issuesAndPullRequests({
+    q: `repo:${owner}/${repo} type:pr state:open 📦 Release @webref/${type}@ in:title`
```

Comment:
Shouldn't the title be quoted here?

And it might be easier to just use two unique branch name patterns and find the PR by branch name.

Reply:
Both good points. I'll change that.

I was under the impression that quotes did not exist in GitHub's search API, but [quotes can be used](https://docs.github.com/en/github/searching-for-information-on-github/understanding-the-search-syntax#use-quotation-marks-for-queries-with-whitespace) indeed.

And I thought that branch name had to be the exact branch name to match, but that is also wrong, searching for `release-idl-` or `release-css-` works fine.